src/targets/gpu/time_op.cpp

```diff
@@ -41,35 +42,58 @@ std::vector<argument> generate_arguments(const std::vector<shape>& shapes, unsig
     return args;
 }
 
-double time_op(context& ictx, operation op, const std::vector<shape>& inputs, int n)
+template <class F>
+double time_loop(migraphx::gpu::context& gctx, int n, F f)
 {
-    // TODO: Use std::ref
-    migraphx::context ctx = ictx;
-    auto& gctx            = any_cast<migraphx::gpu::context>(ctx);
-    auto output           = op.compute_shape(inputs);
-    op.finalize(ctx, output, inputs);
-    auto args = generate_arguments(inputs);
     auto start = context::create_event_for_timing();
     auto stop  = context::create_event_for_timing();
-    auto run   = [&] { op.compute(ctx, output, args); };
-    run();
+    f();
     gctx.get_stream().record(start.get());
     for(auto i : range(n))
     {
         (void)i;
-        run();
+        f();
     }
     gctx.get_stream().record(stop.get());
     gctx.finish();
     return context::get_elapsed_ms(start.get(), stop.get()) / n;
 }
 
-double time_op(context& ictx, operation op, int n)
+double time_op(const context& ictx, operation op, const std::vector<shape>& inputs, int n)
+{
+    // TODO: Use std::ref
+    migraphx::context ctx = ictx;
+    auto& gctx            = any_cast<migraphx::gpu::context>(ctx);
+    auto output           = op.compute_shape(inputs);
+    op.finalize(ctx, output, inputs);
+    auto args = generate_arguments(inputs);
+    auto run  = [&] { op.compute(ctx, output, args); };
+    return time_loop(gctx, n, run);
+}
+
+double time_op(const context& ictx, operation op, int n)
 {
     auto inputs = any_cast<migraphx::gpu::code_object_op>(op).expected_inputs;
     return time_op(ictx, op, inputs, n);
 }
 
+double time_program(const context& ictx, migraphx::program p, int n)
```

Comment:
You can just use `program` directly instead of `migraphx::program`.

Reply:
Done.